test/libyul/yulOptimizerTests/fullSimplify/identity_rules_complex.yul

```diff
@@ -2,7 +2,13 @@
     let a := sub(calldataload(0), calldataload(0))
     mstore(a, 0)
 }
+// ====
+// EVMVersion: >=shanghai
 // ----
 // step: fullSimplify
 //
-// { { mstore(0, 0) } }
+// {
+//     {
+//         mstore(sub(calldataload(0), calldataload(0)), 0)
```

Comment:
This also can't stay like that :-). Which may be an independent issue in the expression simplifier assuming that everything is a variable and also requiring a special case for allowing literals... which might lead back to the same cause as the other cases in the data flow analyzer or might be due to some logic restricting to variables in ``Pattern::matches`` in ``SimplificationRules.cpp```...